controllers/courses.js

```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
 	];
 
 	// ########################### start requests to new Editor #########################
+	let editorBackendIsAlive = true;
 	if (EDITOR_URL) {
-		promises.push(apiEditor(req).get(`course/${req.params.courseId}/lessons`));
+		promises.push(apiEditor(req)
+			.get(`course/${req.params.courseId}/lessons`)
+			.catch((err) => {
+				logger.warn('Can not fetch new editor lessons.', err);
+				editorBackendIsAlive = false;
+				return {
+					total: 0,
+					data: [],
+				};
+			}));
 	}
 
 	// ############################ end requests to new Editor ##########################
+	const [
+		course,
+		_lessons,
+		_homeworks,
+		_courseGroups,
+		scopedPermissions,
+		_newLessons,
+	] = await Promise.all(promises).catch((err) => {
+		logger.warn('Can not fetch course datas', err);
+		next(err);
+	});
 
-	Promise.all(promises)
-		.then(([course, _lessons, _homeworks, _courseGroups, scopedPermissions, _newLessons]) => {
-			// ############################## check if new Editor options should show #################
-			const userHasEditorEnabled = EDITOR_URL && (res.locals.currentUser.features || []).includes('edtr');
-			const courseHasNewEditorLessons = ((_newLessons || {}).total || 0) > 0;
-
-			const isNewEdtrioActivated = (courseHasNewEditorLessons || userHasEditorEnabled);
-			// ################################ end new Editor check ##################################
-
-			const ltiToolIds = (course.ltiToolIds || []).filter(
-				ltiTool => ltiTool.isTemplate !== 'true',
-			).map((tool) => {
-				tool.isBBB = tool.name === 'Video-Konferenz mit BigBlueButton';
-				return tool;
-			});
-			const lessons = (_lessons.data || []).map(lesson => Object.assign(lesson, {
-				url: `/courses/${req.params.courseId}/topics/${lesson._id}/`,
-			}));
-
-			const homeworks = (_homeworks.data || []).map((assignment) => {
-				assignment.url = `/homework/${assignment._id}`;
-				return assignment;
-			});
+	try {
+		// ############################## check if new Editor options should show #################
+		const edtrUser = (res.locals.currentUser.features || []).includes('edtr');
+		const userHasEditorEnabled = EDITOR_URL && edtrUser;
```

Comment:
the meaning is not correct? why doing this if feature is globally disabled?

Reply:
Some closed beta tester can have the flag to use the new editor. If the editor backend is set and alive and a visible editor lesson exist, it should display it to the students.